app/Http/Traits/UploadFileTrait.php

```
@@ -2,30 +2,41 @@
+        $hashname = Str::random(20);
+        $outputFileName = $newName ?? $file->getClientOriginalName();
+        $file->storePubliclyAs($fileLocation . $hashname, $outputFileName, 'public');
+        return $fileLocation . $hashname . "/" . $outputFileName;
+    }
+
+    public function uploadFileStoragedisk($file, $fileLocation, $filename)
+    {
+        if (!file_exists($fileLocation)) {
+            Storage::disk('public')->makeDirectory($fileLocation);
+        }
+        $hashname = Str::random(20);
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Str' in method 'uploadFileStoragedisk'. (undefined)

(StaticAccess)